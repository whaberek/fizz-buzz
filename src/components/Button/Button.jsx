import React from 'react';

const Button = props => (
  <button {...props}>{props.label}</button>
);

export { Button };
