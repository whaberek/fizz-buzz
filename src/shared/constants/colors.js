const colors = {
  black: '#000000',
  green: '#4B8B3B',
  blue: '#2271B3',
};

export { colors };
