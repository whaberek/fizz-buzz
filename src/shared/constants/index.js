export { colors } from './colors';
export { texts } from './texts';
export { validateConstants } from './validateConstants';
