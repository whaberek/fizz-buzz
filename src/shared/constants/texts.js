const texts = {
  fizz: 'fizz',
  buzz: 'buzz',
  wizz: 'wizz',
  wuzz: 'wuzz',
  submit: 'Submit',
  next: 'Next',
  previous: 'Previous',
};

export { texts };
