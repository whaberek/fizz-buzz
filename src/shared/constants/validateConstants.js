const validateConstants = {
  INTEGER: 'Filled value must be an integer.',
  TOO_SMALL: 'Filled value is too small.',
  TOO_HIGH: 'Filled value is too high.',
};

export { validateConstants };
