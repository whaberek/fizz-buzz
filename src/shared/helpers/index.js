export { checkDay } from './checkDay'
export { modulo } from './modulo'
