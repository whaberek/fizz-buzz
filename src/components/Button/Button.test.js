import React from 'react';
import { shallow } from 'enzyme';

import { Button } from './Button';

import { texts } from 'shared/constants';

describe('Button', () => {
  it('should render layout with label', () => {
    const wrapper = shallow(<Button label={texts.submit} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render layout without label', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
