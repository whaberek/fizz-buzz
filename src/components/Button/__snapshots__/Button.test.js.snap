// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render layout with label 1`] = `
<button>
  Submit
</button>
`;

exports[`Button should render layout without label 1`] = `<button />`;
